workflow/controller/controller.go

```diff
@@ -391,8 +399,8 @@ func (wfc *WorkflowController) processNextItem() bool {
 		return true
 	}
 
-	if key, ok = wfc.throttler.Next(key); !ok {
-		log.Warnf("Workflow %s processing has been postponed due to max parallelism limit", key)
+	if !wfc.latch.Pass(un.GetUID(), un.GetResourceVersion()) {
```

Comment:
Stale objects should not pass here.
Sorry for pervious version. I change the code before commit and didn't check the correctness.